### DISCUSSION TITLE
max speed of lego spike prime motors

### DISCUSSION THREAD

**Discussion by crosszay1** [pybricks/support#1874] [General]
by editing the number within the parenthesis we can control how fast the motor can move: `right_motor.run(number)` 
now my question is, what is the maximum number that can be put into here? if I set the speed to 100 and then change it to 500, there is a noticeable difference. but if we set to 100000000 and compare it to 100000000000 there is zero difference. so essentially, at what point does the speed stop changing in relation to the number?

**Comment by laurensvalk** [comment]
It has units of degrees per second. For most motors, the maximum speed is about 1000 deg/s.

To see the default maximum speed, you can display `print(motor.control.limits())`. The first is the speed limit.

> set to 100000000 and compare it to 100000000000 there is zero difference

Rotating at `100000000000` with a 17 cm beam would have the tip rotate at the speed of light. That would be fun, but it is somewhat out of reach for these motors :smile:

**Reply by crosszay1** [reply]
what are the other 2 numbers?

**Reply by BertLindeman** [reply]
> what are the other 2 numbers?

In general if you want to see the doc in code.pybricks.com you can press the book-icon on the right bottom of the panel with the python or the blocks. In this case you might press in the documentation panel on the hamburger icon on the top left and search for `motor.control.limits`  and find in the Motor class this snippet:
```
Parameters
:
speed (Number, deg/s or Number – Maximum speed. All speed commands will be capped to this value.

acceleration Number, deg/s² or Number, mm/s²) – Slope of the speed curve when accelerating or decelerating. Use a tuple to set acceleration and deceleration separately. If one value is given, it is used for both.

torque torque: mNm – Maximum feedback torque during control.
```

Bert

**Comment by afarago** [comment]
Hi Bert,

This is what I found in the official LEGO documentation 

```
velocity: int
The velocity in degrees/sec

Value ranges depends on motor type.

Small motor (essential): -660 to 660
Medium motor: -1110 to 1110
Large motor: -1050 to 1050
```
